atlasdb-config/src/main/java/com/palantir/atlasdb/factory/TransactionManagers.java

```diff
@@ -373,6 +382,15 @@ private TransactionManager serializableInternal(@Output List<AutoCloseable> clos
                 targetedSweep.singleAttemptCallback(),
                 asyncInitializationCallback());
 
+        LoadingCache<TransactionConfig, TransactionConfig> transactionConfigLoadingCache = Caffeine.newBuilder()
```

Comment:
This does seem like an overkill. You can just have a simple static inner class that maintains a reference to the last observed config, and do caching yourself

Reply:
Hmm, unfortunately yeah, this stuff has been requested in several places - https://github.com/google/guava/issues/872 - but there doesn't seem to be nice library support